src/test/java/uk/gov/pay/directdebit/payments/services/PaymentSearchServiceTest.java

```diff
@@ -189,4 +128,22 @@ public void shouldReturnNoRecords_whenPaginationSetToPage2AndNoDisplaySize() {
         assertThat(paymentViewResponse.getLinksForSearchResult().getNextLink(), is(nullValue()));
         assertThat(paymentViewResponse.getLinksForSearchResult().getSelfLink().getHref().contains("?page=2&display_size=500"), is(true));
     }
+
+
+    @NotNull
+    private List<PaymentResponse> createPaymentResponses(int numberOfPayments) {
+        List<PaymentResponse> paymentViewList = new ArrayList<>();
```

Comment:
Another way of doing the same thing using streams, only offering out of interest no need to change.
```
return IntStream.rangeClosed(1, numberOfPayments)
                .mapToObj(i -> aPaymentResponse()
                        .withCreatedDate(createdDate)
                        .withState(new ExternalPaymentStateWithDetails(ExternalPaymentState.EXTERNAL_PENDING, "example_details"))
                        .withReference("Pay reference " + i)
                        .withAmount(1000L + i)
                        .withDescription("This is a description " + i)
                        .withPaymentExternalId(RandomIdGenerator.newId())
                        .build())
                .collect(Collectors.toList());
```